File: lab/python/palindromos/refactory_palinder.py
def normalizar(string):
    """
    Função responsável por editar strings
    """
    return string.lower().replace(" ", "")


def is_palinder(string):
    """
    Verifica se uma string é palíndroma ou não
    """
    if not string:
        return False

    new_string = normalizar(string)

    return new_string == new_string[::-1]


testes = ["", "a", "aa", "ab", "Radar", "a sac a"]
for teste in testes:
    if is_palinder(teste):
        print("SIM")
    else:
        print("NÃO")

File: lab/javascript/palindromos/refactory_palinder.js
function normalizar(string){
    return String(string).toLowerCase().replace(/\s+/g, "");
}


function is_palinder(string){
    if (! string){
        return false;
    }

    let new_string = normalizar(string);

    return new_string === new_string.split("").reverse().join("");
}


let testes = ["", "a", "aa", "ab", "Radar", "a sac a"];
testes.forEach(element => {
    if (is_palinder(element)){
        console.log("SIM");
    } else{
        console.log("NAO");
    }
});
